File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 2

  def self.compute(strand_one, strand_two)
    raise ArgumentError if strand_one.length != strand_two.length

    strand_one.chars.each_with_index.count do |value, index|
      value != strand_two[index]
    end
  end
end

File: scala/hamming/src/main/scala/Hamming.scala
object Hamming {
  def compute(strand_one: String, strand_two: String) : Option[Int] = {
    if(strand_one.length != strand_two.length)
      None
    else
      Some(strand_one.zip(strand_two).count { case (elem_one, elem_two) => elem_one != elem_two})
  }
}

